### DISCUSSION TITLE
Does XState support tokenized event descriptions as proposed in SCXML?

### DISCUSSION THREAD

**Discussion by FadySalama** [statelyai/xstate#3692] [Q&A] [answered]
SCXML states that event descriptors are strings that can be tokenized using dot notation. A transition with an event 'error' for example will match with events such as "error", "error.send", "error.send.failed", etc. Is this supported by XState?

**Comment by Andarist** [comment, marked as answer]
In the upcoming v5 we support this - but we require a wildcard at the end. So `error.*` matches `error`, `error.send` and `error.send.failed` but `error` only matches `error`.

I wonder - do you have an example of how you'd utilize this feature? I was always wondering how one can utilize this in neat ways.

**Reply by FadySalama** [reply]
I do not have a running example, but I can think of scenarios where this is applicable.  For the error example, some machines enter an error state no matter what kind of error is emitted during its operation. Modelling all possible error emissions as an edge to the error state would make the state machine explode with edges. So we capture all possible error events with `error.*` and then inside the error state we transition further based on the specific type of the error. The idea is that it helps minimize the amount of edges that need to be modelled in the system.
